### DISCUSSION TITLE
MSOuath 2.0 Intergration

### DISCUSSION THREAD

**Discussion by jamesinger79** [prasathmani/tinyfilemanager#921] [Q&A]
I would like to implement OAuth 2.0 in this application, but use the permissions for each directory.
Also still using the username and password instead of using the user array.
What would need to be changed

I have MSOuath2 working except on how to display the users folder when redirected back the the application.

Any assistance would be awesome

**Comment by jamesinger79** [comment]
I'm using this to get OAuth2 working - https://learn.microsoft.com/en-us/azure/active-directory/develop/v2-oauth-ropc

**Comment by prasathmani** [comment]
https://www.youtube.com/watch?v=KEX2xenQENc

**Reply by jamesinger79** [reply]
this is for apache setup - cannot se why MS OAUTH2 has to do with  it

**Reply by jamesinger79** [reply]
OK, I see how this is done but not very easy to maintain
i sorted it out my self - had the msoauth do its thing after check if creds and then alow to continue
